src/v/crypto/include/crypto/crypto.h

```diff
@@ -80,4 +80,66 @@ void digest_update(digest_context& ctx, bytes_view msg);
  */
 std::optional<bytes_span<>>
 digest_final(digest_context& ctx, bytes_span<> digest);
+
+///////////////////////////////////////////////////////////////////////////////
+/// MAC operations
+///////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Structure used to hold the context for message authentication code operations
+ */
+struct mac_context {
```

Comment:
nit: maybe there can be one context, using inheritance to make other concrete types from it, or named_type